tests/Core/Content/Product/SalesChannel/Listing/ProductListingFeaturesSubscriberTest.php

```diff
@@ -1107,4 +1113,36 @@ public function testHandleResultDoesNotThrowExceptionWhenCalledManually(): void
         $subscriber = $this->getDefaultProductListingFeaturesSubscriber();
         $subscriber->handleResult($event);
     }
+
+    public function testHandleListingRequestDoesNotThrowAnExceptionWhenCalledManuallyOnANonCategoryPage(): void
+    {
+        $this->configMock->expects($this->any())->method('getShopkey')
+            ->willReturn('ABCDABCDABCDABCDABCDABCDABCDABCD');
+        $this->configMock->expects($this->any())->method('isActive')->willReturn(true);
+        $this->configMock->expects($this->any())->method('isActiveOnCategoryPages')->willReturn(true);
+        $salesChannelContextMock = $this->getMockBuilder(SalesChannelContext::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $eventMock = $this->setUpNavigationRequestMocks(null, [
+            ['findologicService', null],
+            ['flSmartDidYouMean', $this->getDefaultSmartDidYouMeanExtension()]
+        ]);
+        $eventMock->expects($this->any())->method('getSalesChannelContext')
+            ->willReturn($salesChannelContextMock);
+
+        /** @var Request|MockObject $requestMock */
+        $requestMock = $eventMock->getRequest();
+        $requestMock->expects($this->any())->method('get')
+            ->willReturnCallback(function ($param) {
+                if ($param === 'navigationId') {
+                    return Uuid::randomHex();
+                }
+
+                return null;
+            });
+
+        $subscriber = $this->getDefaultProductListingFeaturesSubscriber();
+        $subscriber->handleListingRequest($eventMock);
+    }
```

Comment:
This test doesn't assert anything, is it possible to check if a findologicService has been set to the context?

Reply:
Since `$eventMock` is a mock itself, and we mocked `getExtension`, to already return `null`, I ensured that `addExtension` is properly called with a new instance of `FindologicService`. To ensure this actually happened, I also declared `$addExtensionWasCalled`, which only is being set to `true` if `addExtension` is ever called with the argument `findologicService`. 

Please check out the code changes in this regard.

Reply:
Thanks, this makes it easier to follow. Also when someone accidentally breaks it, easier to fix!